### ISSUE TITLE
Inconsistent path matching between append and parallel

### ISSUE THREAD

**Issue by davidism** [coveragepy/coveragepy#713]
I want to use Coverage to report across multiple Tox envs. Since each Tox env uses a separate virtualenv, I specify `paths` in Coverage's config:

```ini
[coverage:paths]
source =
    src
    .tox/*/lib/python*/site-packages
    .tox/*/site-packages
```

My package is under the src directory. The first tox path applies to CPython envs, the second to PyPy envs, which use a simpler path to site-packages. Specifying the actual package name at the end of each of the paths, or changing the order of the tox lines, or explicitly spelling out each tox path rather than using globs, doesn't affect the issue.

If I use the command `coverage run -p -m pytest` (parallel mode), then add an extra env at the end to do `coverage combine`, I get the correct report, with all tox paths merged into the correct source path. However, if I only want to run one env, like `tox -e py36`, then I end up with stray parallel files since the env that runs `combine` isn't specified.

```
Name                           Stmts   Miss Branch BrPart  Cover
----------------------------------------------------------------
src/itsdangerous/__init__.py     420     68     98     19    82%
tests/test_itsdangerous.py       233      3     27      0    99%
----------------------------------------------------------------
TOTAL                            653     71    125     19    88%
```

I thought I'd use append mode instead of parallel mode: `coverage run -a -m pytest`. This *almost* works, except the pypy path is always listed separately. The *pypy3* path is merged just fine, as are all the CPython paths.

```
Name                                               Stmts   Miss Branch BrPart  Cover
------------------------------------------------------------------------------------
.tox/pypy/site-packages/itsdangerous/__init__.py     420     73     98     23    81%
src/itsdangerous/__init__.py                         420     68     98     19    82%
tests/test_itsdangerous.py                           233      3     27      0    99%
------------------------------------------------------------------------------------
TOTAL                                               1073    144    223     42    85%
```

I expected that paths would be appended to a single file in the same way they're combined from parallel files.

**Comment by nedbat** [comment]
@davidism Thanks, can you provide a way for me to reproduce the failure, even if it's a commit hash in your repo, and commands to run?

**Comment by davidism** [comment]
This was from itsdangerous, but it's consistent across the projects I work on. Here's a minimal example project: [example.tar.gz](https://github.com/nedbat/coveragepy/files/2461398/example.tar.gz)


1. I use Linux, not sure if Windows is different. Need CPython 3.7, CPython 2.7, PyPy 3, PyPy.
1. Extract, create a venv, `pip install -e .` will install the project and tox.
2. `tox -c tox-p.ini` will run coverage in parallel mode, combines and reports files correctly.
3. `tox -c tox-a.ini` will run in append mode, pypy env is reported as separate files.
    
    ```
    Name                                          Stmts   Miss Branch BrPart  Cover
    -------------------------------------------------------------------------------
    .tox/pypy/site-packages/example/__init__.py       0      0      0      0   100%
    .tox/pypy/site-packages/example/encoding.py      21      1      4      1    92%
    src/example/__init__.py                           0      0      0      0   100%
    src/example/encoding.py                          21      0      4      0   100%
    tests/test_example.py                            13      0      0      0   100%
    -------------------------------------------------------------------------------
    TOTAL                                            55      1      8      1    97%
    ```

**Comment by davidism** [comment]
@nedbat was there something else you needed before removing the question tag?

**Comment by nedbat** [comment]
(Sorry, i just haven't come back to look at the answer yet!)

**Comment by nedbat** [comment]
Sorry I've let this sit for so long...

When I run the example (thanks!), I see all the Python interpreters split out separately for tox-a.ini:
```
Name                                                        Stmts   Miss Branch BrPart  Cover
---------------------------------------------------------------------------------------------
.tox/py27/lib/python2.7/site-packages/example/__init__.py       0      0      0      0   100%
.tox/py27/lib/python2.7/site-packages/example/encoding.py      21      1      4      1    92%
.tox/py37/lib/python3.7/site-packages/example/__init__.py       0      0      0      0   100%
.tox/py37/lib/python3.7/site-packages/example/encoding.py      21      1      4      1    92%
.tox/pypy3/site-packages/example/__init__.py                    0      0      0      0   100%
.tox/pypy3/site-packages/example/encoding.py                   21      1      4      1    92%
src/example/__init__.py                                         0      0      0      0   100%
src/example/encoding.py                                        21      0      4      0   100%
tests/test_example.py                                          13      0      0      0   100%
---------------------------------------------------------------------------------------------
TOTAL                                                          97      3     16      3    95%
```

This is because only `coverage combine` re-writes paths.  If you move the .coverage file and run combine, it re-maps the paths to produce the right result.

Issue #1212 proposes to re-map paths without `coverage combine`, which would solve this problem.  I'll close this issue as a duplicate of that one.

**Comment by nedbat** [comment]
This is now fixed in commit 0818611c.

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.0.0b1](https://pypi.org/project/coverage/7.0.0b1).
